src/endpoint-core/Endpoint.Core.Generator/SourceGeneratorExtensions/SourceGeneratorExtensions.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace PrimeFuncPack;
+
+internal static partial class SourceGeneratorExtensions
+{
+    private const string SystemNamespace = "System";
+
+    private const string PrimeFuncPackNamespace = "PrimeFuncPack";
+
+    private const string AsyncSuffix = "Async";
+
+    private const string InnerPrefix = "Inner";
+
+    private static readonly IReadOnlyList<string> DefaultEndpointNameEnds
+        =
+        [
+            "HttpApi", "Api", "HttpFunc", "Func"
+        ];
+
+    private static readonly EndpointOperationResponse.MetadataProvider EmptyResponseMetadataProvider
+        =
+        new(
+            type: new([PrimeFuncPackNamespace], "EndpointEmptyResponseMetadataProvider"),
+            propertyName: "ResponseMetadata");
+
+    private static IEnumerable<T> FilterNotNull<T>(this IEnumerable<T?> source)
+    {
+        foreach (var item in source)
+        {
+            if (item is not null)
+            {
+                yield return item;
+            }
+        }
+    }
+
+    private static bool IsEmptyStructure(this ITypeSymbol type)
+    {
+        return type.IsValueType && type.GetMembers().OfType<IFieldSymbol>().Any(IsNotStatic) is false;
+
+        static bool IsNotStatic(IFieldSymbol field)
+            =>
+            field.IsStatic is false;
+    }
+
+    private static string GetEndpointTypeRootName(this ITypeSymbol endpointType)
+    {
+        var name = endpointType.Name;
+        if (name.Length is not > 1)
+        {
+            return name;
+        }
+
+        if (endpointType.TypeKind is TypeKind.Interface && name.StartsWith("I", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
It might be simplified to following, but let's keep it 'as is' for the sake of uniformity, to correspond other `StartsWith` and similar checks.

```suggestion
        if (endpointType.TypeKind is TypeKind.Interface && name.StartsWith('I'))
```